.github/workflows/api-baseline-generation.yml

```diff
@@ -0,0 +1,194 @@
+name: API Baseline Generation
+
+on:
```

Comment:
Maybe we should add a cron job that regenerates the artifacts every 89 days to avoid them expiring if we dont regenerate them in 90 days?

Reply:
If this runs on _every_ push to main, it's unlikely we need a timed rerun. The manual dispatch is enough to cover the occasional case where everyone is on vacation for 3 months

Reply:
We have a compiler update every 6 weeks, which probably qualifies :)